src/Compilers/CSharp/Portable/Symbols/UnboundGenericType.cs

```diff
@@ -119,4 +119,58 @@ public override int GetHashCode()
                 : Hash.Combine(_name, _errorInfo.Code);
         }
     }
+
+    /// <summary>
+    /// Used for lightweight binding of type constraints. Instead of binding type arguments,
+    /// we'll just use these placeholders instead. That's good enough binding to compute
+    /// <see cref="TypeSymbol.IsValueType"/> with minimal binding.
+    /// </summary>
+    internal sealed class PlaceholderTypeArgumentSymbol : ErrorTypeSymbol
+    {
+        private static readonly TypeWithAnnotations s_instance = TypeWithAnnotations.Create(new PlaceholderTypeArgumentSymbol());
+
+        public static ImmutableArray<TypeWithAnnotations> CreateTypeArguments(ImmutableArray<TypeParameterSymbol> typeParameters)
+        {
+            return typeParameters.SelectAsArray(_ => s_instance);
+        }
+
+        private PlaceholderTypeArgumentSymbol()
+        {
+        }
+
+        protected override NamedTypeSymbol WithTupleDataCore(TupleExtraData newData)
+        {
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return string.Empty;
+            }
+        }
+
+        internal override bool MangleName
+        {
+            get
+            {
+                Debug.Assert(Arity == 0);
+                return false;
+            }
+        }
+
+        internal override DiagnosticInfo? ErrorInfo
+        {
+            get
+            {
+                return null;
+            }
+        }
+
+        internal override bool Equals(TypeSymbol t2, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool>? isValueTypeOverrideOpt = null)
+        {
+            return (object)t2 == this;
```

Comment:
>(object)t2 == this [](start = 19, length = 18)

true?

Reply:
If we returned `true`, then this symbol equals every other symbol.

---
In reply to: [503348621](https://github.com/dotnet/roslyn/pull/48270#discussion_r503348621) [](ancestors = 503348621)

Reply:
> If we returned true, then this symbol equals every other symbol.

Right, I was thinking that all instances of this type should be considered equal to each other (since we have only one). But what I suggested  clearly doesn't do that and it is probably not worth switching to a type check here.

---
In reply to: [503622221](https://github.com/dotnet/roslyn/pull/48270#discussion_r503622221) [](ancestors = 503622221,503348621)